types/part_set.go

```diff
@@ -79,17 +73,18 @@ func (psh PartSetHeader) ValidateBasic() error {
 //-------------------------------------
 
 type PartSet struct {
-	total int32
+	total uint32
 	hash  []byte
 
 	mtx           sync.Mutex
 	parts         []*Part
 	partsBitArray *bits.BitArray
-	count         int32
+	count         uint32
 }
 
 // Returns an immutable, full PartSet from the data bytes.
 // The data bytes are split into "partSize" chunks, and merkle tree computed.
+// CONTRACT: partSize is greater than zero.
 func NewPartSetFromData(data []byte, partSize int) *PartSet {
```

Comment:
Should we change partSize to be an uint64/32 too?

Reply:
is it better to cast to uint32 or to int?

Reply:
If the number should never be negative, I'd say it should be a uint. And sized is better than usized, since `uint` will be either `uint32` or `uint64` depending on the hardware.